src/io/parquet/write/mod.rs

```diff
@@ -163,8 +166,31 @@ pub fn array_to_pages(
                     )
                 })
             }
-            _ => array_to_page(array, type_, nested, options, encoding)
-                .map(|page| DynIter::new(std::iter::once(Ok(page)))),
+            _ => {
+                const DEFAULT_PAGE_SIZE: usize = 1024 * 1024;
+                let page_size = options.data_pagesize_limit.unwrap_or(DEFAULT_PAGE_SIZE);
+
+                let bytes_per_row = estimated_bytes_size(array.slice(0, 1).as_ref()).max(1);
```

Comment:
```
let bytes_per_row = ((estimated_bytes_size(array.as_ref()) as f64) / array.len()).max(1); // needs some rounding to integer
```

likely produces more reliable estimates for variable sized arrays, as the first element may be an outlier.